pkg/controller/flightctl/flightctl.go

```diff
@@ -0,0 +1,186 @@
+package flightctl
+
+import (
+	"context"
+	"embed"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+
+	flightctlapiv1 "github.com/flightctl/flightctl/lib/apipublic/v1alpha1"
+	flightctlcli "github.com/flightctl/flightctl/lib/cli"
+	"github.com/stolostron/managedcluster-import-controller/pkg/helpers"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	FlightCtlNamespace      = "flightctl"
+	FlightCtlInternalServer = "https://flightctl-api.flightctl.svc.cluster.local:3443"
+)
+
+//go:embed manifests
+var FlightCtlManifestFiles embed.FS
+
+// The flightctl-client's service account token has 2 usages:
+// 1. delivered to the devices, used to access the agent-registration to get klusterlet manifests used for registration.
+// 2. on the hub side, used for the import-controller to apply the flightctl's Repository resources and get devices.
+var files = []string{
+	"manifests/clusterrole.yml",
+	"manifests/clusterrolebinding_agentregistration.yml",
+	"manifests/clusterrolebinding_flightctl.yml",
+	"manifests/serviceaccount.yml",
+	"manifests/serviceaccounttokensecret.yml",
+	"manifests/networkpolicy.yml",
+}
+
+type FlightCtler interface {
+	IsFlightCtlEnabled(ctx context.Context) (bool, error)
+	ApplyRepository(ctx context.Context) error
+	IsManagedClusterAFlightctlDevice(ctx context.Context, managedClusterName string) (bool, error)
+}
+
+func NewFlightCtl(clientHolder *helpers.ClientHolder, clusterIngressDomain string) FlightCtler {
+	return &FlightCtl{
+		flightctlServer:         FlightCtlInternalServer,
+		agentRegistrationServer: "https://agent-registration-multicluster-engine." + clusterIngressDomain,
+		clientHolder:            clientHolder,
+	}
+}
+
+type FlightCtl struct {
+	clientHolder            *helpers.ClientHolder
+	flightctlServer         string
+	agentRegistrationServer string
+	cachedToken             string
+	cachedCA                string
+}
+
+var _ FlightCtler = &FlightCtl{}
+
+func (f *FlightCtl) IsFlightCtlEnabled(ctx context.Context) (bool, error) {
+	// If `flightctl` ns is not created, means the flightctl is not enabled.
+	ns := &corev1.Namespace{}
```

Comment:
if we already have the server url, why not call server to check if it is enabled?